### PR TITLE
feat(NODE-5054): add AssumeRoleWithWebIdentity support to 4x driver

### PR DESCRIPTION
**Pull request by baileympearson**
### Description

This PR backports testing for AWS EKS requests.

##### Is there new documentation needed for these changes?

No.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [x] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -649,6 +654,53 @@ functions:
         script: |
           ${PREPARE_SHELL}
           ${PROJECT_DIRECTORY}/.evergreen/run-mongodb-aws-test.sh
+  run aws auth test AssumeRoleWithWebIdentity with AWS_ROLE_SESSION_NAME set:
```
Comment (dariakp) [inline .evergreen/config.yml:657]: Noticed that this set of changes doesn't have the addition of the `install aws-credential-providers` script that #3556 did - why the discrepancy?

File: package.json [hunk 2]
```diff
@@ -136,4 +136,4 @@
       "moduleResolution": "node"
     }
   }
-}
+}
```
Comment (dariakp) [inline package.json:139]: why was this file touched?

File: .evergreen/config.yml (hunk 1 above)
Comment (baileympearson) [reply .evergreen/config.yml:657]: In version 4x, the AWS SDK was an optional dependency so it was installed by default. In version five, it is no longer installed by default so we must manually install it.

File: package.json (hunk 2 above)
Comment (baileympearson) [reply package.json:139]: Just a bad merge. I'll fix it

**Pull request merged**